src/adaptation/icap/Xaction.cc

```diff
@@ -741,7 +741,7 @@ Adaptation::Icap::Xaction::handleSecuredPeer(Security::EncryptorAnswer &answer)
     securer = NULL;
 
     if (closer != NULL) {
-        if (answer.conn != NULL)
+        if (answer.conn != NULL && answer.conn->isOpen())
```

Comment:
Please do not use NULL in new or changed lines.

```suggestion
        if (answer.conn && answer.conn->isOpen())
```

Reply:
or better use the Comm::IsConnOpen test:

```suggestion
        if (Comm::IsConnOpen(answer.conn))
```

Reply:
I used the second form.